### PR TITLE
feat: add paymaster guard module

### PR DESCRIPTION
**Pull request by linnall**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing the `PaymasterGuardModule` functionality by adding installation and uninstallation processes, as well as testing scenarios for valid and invalid paymaster usage within the `installValidation` actions.

### Detailed summary
- Commented out error handling for `DEFAULT_OWNER_ENTITY_ID` in `installValidation.ts`.
- Introduced `PaymasterGuardModule` with `encodeOnInstallData` and `encodeOnUninstallData` methods in `module.ts`.
- Updated tests in `client.test.ts` to include scenarios for installing and uninstalling the `PaymasterGuardModule`.
- Verified correct paymaster usage and handled cases for invalid paymaster in the tests.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 1]
```diff
@@ -182,17 +192,181 @@ describe("MA v2 Tests", async () => {
...
+          },
+          initData: hookInstallData,
+        },
+      ],
+    });
+
+    // verify hook installtion succeeded
+    await expect(
+      provider.waitForUserOperationTransaction(installResult)
+    ).resolves.not.toThrowError();
```
Comment (dphilipson) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:242]: This is fine, but note that `.resolves.not.toThrowError()` isn't particularly better than just awaiting the promise, which will also fail the test if the promise fails, e.g.
```ts
await provider.waitForUserOperationTransaction(
  installResult,
);
```

File: account-kit/smart-contracts/src/ma-v2/modules/paymaster-guard-module/module.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { encodeAbiParameters, type Address, type Hex } from "viem";
+import { paymasterGuardModuleAbi } from "./abis/paymasterGuardModuleAbi.js";
+
+const addresses = {
+  default: "0x00", //dummy address, replace with deployed module address?
```
Comment (dphilipson) [inline account-kit/smart-contracts/src/ma-v2/modules/paymaster-guard-module/module.ts:5]: If you want the zero address, you can import `zeroAddress` from viem. I'm not sure that the viem APIs that accept addresses will work with `"0x00"`.

File: account-kit/smart-contracts/src/ma-v2/modules/paymaster-guard-module/module.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  encodeOnInstallData: (args: {
+    entityId: number;
+    paymaster: Address;
+  }): Hex => {
+    const { entityId, paymaster } = args;
+    return encodeAbiParameters(
+      [
+        {
+          type: "uint32",
+          value: entityId,
```
Comment (dphilipson) [inline account-kit/smart-contracts/src/ma-v2/modules/paymaster-guard-module/module.ts:26]: `value` isn't a field in the `encodeAbiParameters` API. Each of these elements should have the fields `type` and optionally `name`, where I think `name` is only used for error messages. See [the docs](https://viem.sh/docs/abi/encodeAbiParameters).

**Review by dphilipson — COMMENTED** [review]
Looks good! Some minor comments inline.

**Review by dphilipson — APPROVED** [review]

File: account-kit/smart-contracts/src/ma-v2/modules/paymaster-guard-module/module.ts (hunk 2 above)
Comment (linnall) [reply account-kit/smart-contracts/src/ma-v2/modules/paymaster-guard-module/module.ts]: `addresses` is no longer being used in the module definition, resolving this comment!

**Pull request merged**